tests/test_types.py

```diff
@@ -1526,3 +1526,47 @@ class Foobar(BaseModel):
     with pytest.raises(ValidationError) as exc_info:
         Foobar(password=[6, 23, 'abc'])
     assert exc_info.value.errors() == [{'loc': ('password',), 'msg': 'byte type expected', 'type': 'type_error.bytes'}]
+
+
+def test_generic_without_params():
+    class Model(BaseModel):
+        generic_list: List
+        generic_dict: Dict
+
+    m = Model(generic_list=[0, 'a'], generic_dict={0: 'a', 'a': 0})
+    assert m.dict() == {'generic_list': [0, 'a'], 'generic_dict': {0: 'a', 'a': 0}}
+
+
+def test_generic_without_params_error():
+    class Model(BaseModel):
+        generic_list: List
+        generic_dict: Dict
+
+    with pytest.raises(ValidationError) as exc_info:
+        Model(generic_list=0, generic_dict=0)
+    assert exc_info.value.errors() == [
+        {'loc': ('generic_list',), 'msg': 'value is not a valid list', 'type': 'type_error.list'},
+        {'loc': ('generic_dict',), 'msg': 'value is not a valid dict', 'type': 'type_error.dict'},
+    ]
+
+
+# THIS TEST WILL BE REMOVED BEFORE FINAL PULL REQUEST SUBMISSION
+def test_is_this_okay():  # ???
+    class Model(BaseModel):
+        generic_list: List
+        generic_dict: Dict[int, int]
+
+    # This test passes -- should it fail, or is this the desired behavior? (It would make sense to me either way)
+    # It also passes if I replace Dict with Dict[int, int], so no new problems have been introduced.
+    assert Model(generic_list=[], generic_dict=[])
```

Comment:
This should definitely fail with the error as below.

However I've just check and it's already wrong on master so should be fixed separately. #558